devops/clean

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+# Cleans all build artifacts and custom config from the repo.
+# Intended for use on developer workstations. Should NOT be run
+# on production Admin Workstations.
+set -e
+set -u
+
+
+function remove_unwanted_files() {
+
+    # Remove any Onion URL info from previous instances.
+    rm -vf install_files/ansible-base/app-source-ths \
+        install_files/ansible-base/app-ssh-aths \
+        install_files/ansible-base/app-journalist-aths \
+        install_files/ansible-base/mon-ssh-aths \
+        build/*.deb
+
+    # Remove extraneous copies of the git repos, pulled in
+    # via the Molecule upgrade testing scenario.
+    rm -rf molecule/upgrade/.molecule/sd-orig \
+        molecule/vagrant_packager/.molecule/sd-orig
+
+    # Python bytecode, left over from tests or local app runs.
+    find -type f -iname '*.pyc' -delete
+
+    # Ansible playbook retry files, which are never used.
+    find -type f -iname '*.retry' -delete
+
+    # Static assests generated by local dev, as well as
+    # virtualenv for securedrop-admin script.
+    rm -rvf securedrop/static/.webassets-cache .venv
+
+    # Any Vagrant VMs should be destroyed. We check for the .vagrant/
+    # directory, because it'll be created by vagrant otherwise, breaking
+    # idempotence for the cleanup action.
+    if hash vagrant &> /dev/null && [[ -d .vagrant/ ]]
+    then
+        printf "Removing vagrant VMs\\n"
+        vagrant destroy -f > /dev/null
+    fi
+
+    # Any and all git-ignored files and directories, if present. Includes e.g.
+    # deb packages in build/, and VM information stored locally.
+    git clean -X -d -f
+}
+
+
+function confirmation_prompt() {
+    # Display scary warning as last chance to avoid destructive action.
+    cat <<-WARNING_MESSAGE
+WARNING: Proceeding will destroy all local customizations, including
+application configuration, user credentials, Onion URLs, static assets,
+VMs, and all git-ignored files. It is NOT possible to undo these changes!
+WARNING_MESSAGE
+
+    # We want to avoid mistakes here, so force developers
+    # to type 'yes' in order to confirm and proceed with cleanup.
+    read -r -p "To proceed, type 'yes': " user_confirmation
+    printf "\\n"
+    case "${user_confirmation}" in
+        yes ) remove_unwanted_files;;
+        * ) printf "Action declined, exiting...\\n"
```

Comment:
I would vote for an `exit 1` here.

Reply:
Also extra backslash?

Reply:
Motivated by https://github.com/koalaman/shellcheck/wiki/SC1117, but I agree it's confusing. Will revise.

Reply:
Hmm I guess either way works then? Damn `bash` is so weird. I swear I will never fully understand it.